File: Automation/TestAutomation/AppiumCore/Base/ScreenFactory.cs
using System;

namespace AppiumCore.Base
{
    public class ScreenFactory
    {
        private static Lazy<ScreenFactory> _instance = new Lazy<ScreenFactory>(() => new ScreenFactory());

        public static ScreenFactory Instance
        {
            get
            {
                return _instance.Value;
            }
        }

        private ScreenFactory() { }

        public BaseScreen CurrentPage { get; set; }
    }
}

File: Automation/Bosch/TestAutomation/src/main/java/base/ScreenFactory.java
package base;

public class ScreenFactory {
    private static ScreenFactory instance = null;
    public BaseScreen CurrentPage;

    public static ScreenFactory getInstance() {
        if (instance == null){
            instance = new ScreenFactory();
        }

        return instance;
    }

    private ScreenFactory() { }

}
